### PR TITLE
Adding code for spark-ml jobserver post

### PR DESCRIPTION
**Pull request by dgraeber**

### CODE REVIEW

File: aws-blog-jobserver-emr/BA/install_jobserver_ML_BA.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    sudo yum install git -y
+
+    sudo mkdir /mnt/work -p
+    cd /mnt/work
+    sudo git clone https://github.com/spark-jobserver/spark-jobserver.git
+    cd spark-jobserver
+    sudo git checkout $JOBSERVER_VERSION
+    sudo sbt clean update package assembly
+
+
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/BA/install_jobserver_ML_BA.sh]: nit: can you make the spacing consistent?

File: aws-blog-jobserver-emr/BA/install_jobserver_ML_BA.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    #Stage and untar the distro
+    cd /mnt/lib/spark-jobserver
+    sudo tar zxf /tmp/job-server/job-server.tar.gz
+
+    sudo chown hadoop:hadoop /mnt/lib/spark-jobserver -R
+fi
+exit 0
+
+#Start the server
+#./server_start.sh
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/BA/install_jobserver_ML_BA.sh]: extra comments?

File: aws-blog-jobserver-emr/cfn/ml-jobserver.template [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+    },
+
+    "sgIngressLambdaToMaster": {
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "GroupId": {"Ref": "sgEMRMasterAdditional"},"IpProtocol": "-1", "FromPort": "0", "ToPort": "65535", "SourceSecurityGroupId":{"Ref": "sgLambda"}
+      }
+    },
+
+
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/cfn/ml-jobserver.template]: nit: consistent spacing

File: aws-blog-jobserver-emr/html/mltest.html [hunk 4]
```diff
@@ -0,0 +1,284 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <style>
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/html/mltest.html]: will be cleaner to extra css and js into separate files

File: aws-blog-jobserver-emr/python_lambda/models.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+import logging
+import json
+
+from sjsclient import client
+
+logging.basicConfig()
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+#sjs = client.Client("http://ec2-54-172-7-213.compute-1.amazonaws.com:8090")
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/python_lambda/models.py]: please remove commented out code

File: aws-blog-jobserver-emr/python_lambda/models.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    #conf = '{'+'s3DataLoc:{},s3ModelLoc:{}'.format(s3DataLoc,s3ModelLoc)+'}'
+
+    s3DataLocProtocol = "\"s3://{}\"".format(event['s3DataLoc'])
+    s3ModelLocProtocol = "\"s3://{}\"".format(event['s3ModelLoc'])
+    conf = '{'+'s3DataLoc:{},s3ModelLoc:{}'.format(s3DataLocProtocol,s3ModelLocProtocol)+'}'
+
+    class_path = "com.amazonaws.proserv.ml.TestParams"
+    myJob = sjs.jobs.create(myApp, class_path, conf = conf, ctx = myContext)
+    myId = myJob.jobId
+    while myJob.status != "FINISHED":
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/python_lambda/models.py]: would be more readable and maintainable to define constants for hard coded values

File: aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+import com.typesafe.config.Config
+import org.apache.spark.SparkContext
+import spark.jobserver._
+import org.apache.spark.mllib.recommendation.{ALS, Rating, MatrixFactorizationModel}
+
+import scala.collection.mutable.ListBuffer
+
+
+/**
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala]: please remove. this info is in git

File: aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+  override def runJob(sc: SparkContext, jobConfig: Config): Any = {
+
+    val movieLensHomeDir = jobConfig.getString("s3DataLoc")
+    val modelHomeDir = jobConfig.getString("s3ModelLoc")
+
+// Load the movies data (movies and movies with genre
+
+    val movies = sc.textFile(movieLensHomeDir + "movies.dat").map { line =>
+      val fields = line.split("::")
+      // format: (movieId, movieName)
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala]: please remove commented code. makes the code hard to understand

File: aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+    var recs = new ListBuffer[String]()
+    var i = 1
+    recommendations.foreach { r =>
+      recs += "%2d".format(i) + ": " + movies(r.product)
+      i += 1
+    }
+
+    recs.toList
+
+
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala]: nit: spacing

File: aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+  override def validate(sc: SparkContext, config: Config): SparkJobValidation = SparkJobValid //not really checking here...but probably should
+}
+
+object TestParams extends SparkJob with NamedRddSupport {
+  override def runJob(sc: SparkContext, jobConfig: Config): Any = {
+    val movieLensHomeDir = jobConfig.getString("s3DataLoc")
+    val modelHomeDir = jobConfig.getString("s3ModelLoc")
+
+    val res = "Got "+movieLensHomeDir+" and "+modelHomeDir
+    res
```
Comment (mbeitchman) [inline aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala]: don't think the res is needed here

File: aws-blog-jobserver-emr/html/mltest.html (hunk 4 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/html/mltest.html]: Agreed - but for this example I'd like to keep it in a single file as it is not the primary focus of the blogpost / codebase.

File: aws-blog-jobserver-emr/BA/install_jobserver_ML_BA.sh (hunk 1 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/BA/install_jobserver_ML_BA.sh]: Agreed - correcting

File: aws-blog-jobserver-emr/BA/install_jobserver_ML_BA.sh (hunk 2 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/BA/install_jobserver_ML_BA.sh]: Agreed - correcting

File: aws-blog-jobserver-emr/cfn/ml-jobserver.template (hunk 3 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/cfn/ml-jobserver.template]: Agreed - correcting

File: aws-blog-jobserver-emr/python_lambda/models.py (hunk 5 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/python_lambda/models.py]: Agreed - correcting

File: aws-blog-jobserver-emr/python_lambda/models.py (hunk 6 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/python_lambda/models.py]: I have the variables scoped based on usage.  There is some overlap, but fir highlighting the usage for illustration purposes, I'd like to leave as is (but totally agree with your evaluation).

File: aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala (hunk 10 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala]: Yes - it is needed to return the value to the caller

File: aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala (hunk 8 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala]: Agreed - correcting

File: aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala (hunk 9 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala]: Agreed - correcting

File: aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala (hunk 7 above)
Comment (dgraeber) [inline aws-blog-jobserver-emr/src/main/scala/com/amazonaws/proserv/ml/MovieRecML.scala]: Agreed - correcting

**Comment by mbeitchman** [comment]
Looks good. thanks

**Pull request merged**